pkg/kubelet/cm/devicemanager/manager_test.go

```diff
@@ -40,25 +40,42 @@ import (
 )
 
 const (
-	socketName       = "/tmp/device_plugin/server.sock"
-	pluginSocketName = "/tmp/device_plugin/device-plugin.sock"
 	testResourceName = "fake-domain/resource"
 )
 
+func tmpSocketDir() (socketDir, socketName, pluginSocketName string, err error) {
+	socketDir, err = ioutil.TempDir("", "device_plugin")
```

Comment:
Would it be better to get a unique directory here using timestamp based suffix.
```golang
tmpDir, err := ioutil.TempDir("", "device_plugin"+fmt.Sprintf("%d", time.Now().Unix()))
```
And then use this in the setup(). Since there can be several setup() invocations in parallel and setup() calls `m.Start()` which eventually invokes removeContents, setup() should pass unique socket directory to the NewManagerImpl().

Reply:
**Tempdir:**
TempDir creates a new temporary directory in the directory dir with a name beginning with prefix and returns the path of the new directory. If dir is the empty string, TempDir uses the default directory for temporary files (see os.TempDir). Multiple programs calling TempDir simultaneously will not choose the same directory. It is the caller's responsibility to remove the directory when no longer needed.